tools/notification-configuration/notification-creator/NotificationConfigurator.cs

```diff
@@ -28,54 +28,50 @@ public NotificationConfigurator(AzureDevOpsService service, ILogger<Notification
         }
 
         public async Task ConfigureNotifications(
-            string projectName, 
-            string projectPath, 
-            bool persistChanges = true, 
+            string projectName,
+            string projectPath,
+            bool persistChanges = true,
             PipelineSelectionStrategy strategy = PipelineSelectionStrategy.Scheduled)
         {
             var pipelines = await GetPipelinesAsync(projectName, projectPath, strategy);
             var teams = await service.GetAllTeamsAsync(projectName);
 
             foreach (var pipeline in pipelines)
             {
-                // If the pipeline name length is max or greater there is no
-                // room to add differentiators like "-- Sync Notifications"
-                // and this will result in team name collisions.
-                if (pipeline.Name.Length >= MaxTeamNameLength)
-                {
-                    throw new Exception($"Pipeline Name outside of character limit: Max = {MaxTeamNameLength}, Actual = {pipeline.Name.Length}, Name = {pipeline.Name}");
-                }
-
                 using (logger.BeginScope("Evaluate Pipeline Name = {0}, Path = {1}, Id = {2}", pipeline.Name, pipeline.Path, pipeline.Id))
                 {
-                    var parentTeam = await EnsureTeamExists(pipeline, "Notifications", TeamPurpose.ParentNotificationTeam, teams, persistChanges);
-                    var childTeam = await EnsureTeamExists(pipeline, "Sync Notifications", TeamPurpose.SynchronizedNotificationTeam, teams, persistChanges);
+                    var parentTeam = await EnsureTeamExists(pipeline, "[ntfy]", TeamPurpose.ParentNotificationTeam, teams, persistChanges);
+                    var childTeam = await EnsureTeamExists(pipeline, "[sync]", TeamPurpose.SynchronizedNotificationTeam, teams, persistChanges);
```

Comment:
```suggestion
                    var childTeam = await EnsureTeamExists(pipeline, "[owners]", TeamPurpose.SynchronizedNotificationTeam, teams, persistChanges);
```

Reply:
This prefix solution still kicks the can down the road though, the moment we have a pipeline name with a few more characters we're going to hit this problem yet again. The only other solutions I can think of is to not rely on preserving the pipeline name or having some shorthand for the language/test type/separator.

Reply:
After chatting I think we landed on changing this to something like:
1. notify team should match the name of the pipeline. If the name is too long then truncate and add pipeline id at the end to make sure we don't have conflicts.
2. sync team we should just use the pipeline id and use something like "codeowners sync for <id>"

Reply:
Implementing ^^ but I'm putting the pipeline ID at the start of the name, that way the parent and sync notification teams still sort next to each other in the list view for ease of use.

Parent name format: `[<pipeline id>] <pipeline name>]`
Sync name format: `[<pipeline id>] Code owners sync notifications`

Dry run log output example for the new change (handling renaming of existing pipelines):

```
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Found team with matching pipeline id 1034 but different name 'python - sql -- Notifications', expected '[1034] python - sql'. Purpose = 'ParentNotificationTeam'
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Update Team for Pipeline PipelineId = 1034 Purpose = ParentNotificationTeam Name = '[1034] python - sql'
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Found team with matching pipeline id 1034 but different name 'python - sql -- Sync Notifications', expected '[1034] Code owners sync notifications'. Purpose = 'SynchronizedNotificationTeam'
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Update Team for Pipeline PipelineId = 1034 Purpose = SynchronizedNotificationTeam Name = '[1034] Code owners sync notifications'
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      GetDescriptor Id = fb8f4d34-1636-439c-a957-c14881afa7bf
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      GetDescriptor Id = e64231d0-8547-4e65-b61c-bf91456affda
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]                                                                                                                                                                                                                                                                                                                                                               => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      CheckMembership ParentId = vssgp.Uy0xLTktMTU1MTM3NDI0NS03MjEyMjY4NDEtNDc1NTgxMjYxLTI3MDQxNzIzOTItMzQ2Mzk5NTc2NS0xLTg2OTkyODc4Mi0zMDgwNjE3Mjg4LTI2NzI1MjM3NjItMTgxMDM3MDM5NQ ChildId = vssgp.Uy0xLTktMTU1MTM3NDI0NS03MjEyMjY4NDEtNDc1NTgxMjYxLTI3MDQxNzIzOTItMzQ2Mzk5NTc2NS0xLTM4Mjk3Njg0NjItMzk4NTU3MDExNS0yOTI5NzEwMDIyLTIyOTY1ODc1NzE
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Child In Parent ParentId = fb8f4d34-1636-439c-a957-c14881afa7bf, ChildId = e64231d0-8547-4e65-b61c-bf91456affda, IsInTeam = True
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      GetSubscriptionsAsync TargetId = fb8f4d34-1636-439c-a957-c14881afa7bf
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Team Is Subscribed TeamName = [1034] python - sql PipelineId = 1034
```